### PR TITLE
Create 013-ryan.md

### PR DESCRIPTION
**Pull request by amanda-onevariable**
Hi! I did this last week, but apparently forgot to do a PR. There were a couple of names or terms in the transcript I couldn't Google myself out of.. but other than that, the podcast should be ready / this page ready for the embed links

### CODE REVIEW

File: content/podcast/013-ryan.md [hunk 1]
```diff
@@ -0,0 +1,400 @@
...
+
+Cool. So before we get too far with all of that — because I want to talk about all of that — Do you want to give yourself a quick introduction?
+
+*Ryan Summers*
+
+Yeah, my name's *Ryan Summers* I have been doing embedded rust for 4, 5 years now something around that. I do embedded consulting and also have a startup doing embedded manufacturing automation. So if you ever want to look up forged.dev.
+
+*James Munns*
+
+Oh you're working — Oh that's right, you have mentioned that you're working with Noah…
```
Comment (amanda-onevariable) [inline content/podcast/013-ryan.md:157]: Noah?

File: content/podcast/013-ryan.md [hunk 2]
```diff
@@ -0,0 +1,400 @@
...
+
+So there's many things we could talk about today if you want.
+
+*James Munns*
+
+So you mentioned Stabilizer and Booster. Do you want to give a quick example of what each of those are, and how those typically get used for scientific things?
+
+*Ryan Summers*
+
+Yeah, the easier one to start with is definitely Booster. The general idea is that you've got some kind of signal source that you're going to be using to drive some laser in a lab, and generally the laser needs to be running in some kind of high power, and your signal generator doesn't generate high power and so the idea is run coaxial into this thing, specify how much gain you want on the output and it handles all of the power management and generates the, the necessary gain so that you can actually drive your big laser load. And it's got all kinds of nice safety protection mechanisms. It's got fans in there that you can automatically spin up and keep things cool. It'll handle interlock tripping over power and all kinds of fancy things.
```
Comment (amanda-onevariable) [inline content/podcast/013-ryan.md:177]: coaxial
interlock (once more later)

File: content/podcast/013-ryan.md [hunk 3]
```diff
@@ -0,0 +1,400 @@
...
...i or something like that and it's like... If you have a device that's doing TLS and doing all of this, why do you need a lightweight message protocol versus something like ZeroMQ or RabbitMQ or, like, any of the other PubSub message brokers, which get used a lot for backend services for either like control plane. I came to you and I reached out to you because you were saying very positive things about MQTT and I was like, I have this big chip on my shoulder from doing a lot of one-off demos where it worked and it's, like, a reasonable protocol and it has very, like, straightforward nice to work with semantics. But it always bugged me as someone who has done a lot of like — either really hard, real-time embedded stuff, or very like “Ah, we've got 8 kilobytes of RAM” kind of thing, when people describe it as like a lightweight embedded protocol and I'm like — as soon as you have TLS like you've sort of left the realm of lightweight protocol. But I guess since your control plane is largely wired ethernet and you're on I'm hoping a fairly well-segmented network… These aren't connected-to-the-internet sort of devices, these are connected-to-local-control-land kind of devices, hopefully.
```
Comment (amanda-onevariable) [inline content/podcast/013-ryan.md]: I did google NQC, but wasn't confident in the actual search results? Like they were from companies

File: content/podcast/013-ryan.md [hunk 4]
```diff
@@ -0,0 +1,400 @@
...
...some payload and said like, “Hey, set the setting to this value.” And really, we're leveraging Serde at that point where we say like, “Hey, try and turn that into the type we want.” And what's really nice is you can start just propagating errors out using just the `?` and levering things go all the way back. But then you can start catching them at the MQTT interface and start using that error formatting, and then print that back as a response over MQTT. And so suddenly, on your Python side or on your PC where you're trying to configure it when someone tries to set a setting, you immediately get a response stream, and says like, “Hey, that deserialization failed because I expected this, like, `/` and this location we didn't get it.” And so it's really cool to be able to see, like, proper error handling written and code immediately translate into this wonderful user experience on the tooling side, but it it also makes it more complicated because as you're writing a library, you're like, “I need to make these errors right, because this propagates all the way back out to the end user.” It's not just code at this point. Someone needs to be able to read this and understand how to fix it.
```
Comment (amanda-onevariable) [inline content/podcast/013-ryan.md]: was at a loss re: "Robert refactoring minicon"

File: content/podcast/013-ryan.md [hunk 5]
```diff
@@ -0,0 +1,400 @@
...
...nflux so that we can start visualizing our dashboard. We've also got our development tools up there looking at it so we can see like, “Ah, what's the state of the devices we're like running? These control loops and stuff” and so really, you get both things which is really nice. Back in my university days, I did a lot of work with ROS the Robot Operating System and that uses the traditional PubSub methodology. And really like that being able to publish data somewhere makes developing your system a lot simpler because you no longer have to think about how tightly structured it is, you just start blasting out the data for everyone to listen to. Maybe someone will need it and then when you're going and working on some other component in the future, you’re like, “Ah yeah, let me just subscribe to that, because I actually, I could use this specific piece of information and I'm going to go and implement my new functionality based on that.” And so really, you don't have to go and modify that original component you made now to get that data because it's just out there in the open for you. So I think really that the benefits of that PubSub style — it's really just flexibility in your design.
```
Comment (amanda-onevariable) [inline content/podcast/013-ryan.md]: "into a db and influx" ?

File: content/podcast/013-ryan.md [hunk 6]
```diff
@@ -0,0 +1,400 @@
...
...up is they had that cordup does more than just Booster and Stabilizer, there's all kinds of lab rate hardware they're building for people and one of the setups they had was a neutral atom trap and I believe they actually had Stabilizer or one of their other controllers talking to another device, both of which were running MQTT on these mini comp clients. And basically, one was setting an interlock for the other and saying like — hey, disable yourself now based on its measurements, so it was performing like remote temperature sensing, remotely disabling the other device. To me, that was just really cool to see in the end, because you don't set out with the intent to build these kind of systems. When you're like, “Oh, I want to build an MQTT client. Like you have no idea what it's going to be used for. But then when you see someone coming afterwards and like start connecting all these things together across multiple embedded devices and doing end to end communication like — you see that as buzzword all the time, you're like yeah but when did people actually do that and then suddenly like someone did it. It actually helped to solve their problem and it’s like whoa. That's really cool.
```
Comment (amanda-onevariable) [inline content/podcast/013-ryan.md]: H7 series (regarding the STN32H7?) 
Richard Cheung (?)
ABC update
mini comp client

File: content/podcast/013-ryan.md [hunk 7]
```diff
@@ -0,0 +1,400 @@
...
...en. And you get those differentials which are great but it means that you have to think about the next level of complexity up because you go, “Have I heard this message from this device ever?” or if you reboot your tooling loop but not your embedded system, you have to query it. “Give me your your status because I just woke up and I've never heard it before even though you sent it 5 minutes ago, I've never heard it,” or something like that versus like the really dumb world of you just send the whole thing out all the time - no thinking only sending. You never have to think about that because you just go, “Okay. If I just listen then I know everyone's state and the first time I hear from them, I know that they're alive and awake.” But from an efficiency perspective, that's a terrible idea, and if you were to doing anything with battery life: terrible idea. Even for you, where you want to probably reserve as much bandwidth on your network for raw data samples and something like that: control data is probably not a huge fraction of that. But if you just have to spend the time serializing and deserializing those messages, that's time you could be spent doing DSP or something like that.
```
Comment (amanda-onevariable) [inline content/podcast/013-ryan.md:279]: I2C, I^2C, IIC?

File: content/podcast/013-ryan.md [hunk 8]
```diff
@@ -0,0 +1,400 @@
...
...t word. Okay, now we got to go do some ABC, like do some DSB stuff.” It's really interesting to see that. One of the really interesting parts is also when it comes down to real-time in embedded is in your codebase when you're writing functions and you have an edge case where you can say like, “Oh, yeah, we could return early here. We could avoid this expensive calculation.“ Sometimes it's actually better to still do the expensive dumb calculation for no reason, because what it gives you is this known timing characteristic. Suddenly, when you've got this known timing constant, you don't have these weird conditions where most of the time when you're calling this function it works in a millisecond and then sporadically like, garbage collector comes around and it's 99 milliseconds all of a sudden. If you can keep your control loop consistent, you can kind of manage it better, and so sometimes the best path is not the most optimal execution path and you'll say like, even though we could break out early here we're intentionally not doing that. I don't know if that related too much to your earlier question about kind of there quest response stuff, but it kind of jogged an idea in my head.
```
Comment (amanda-onevariable) [inline content/podcast/013-ryan.md]: scatiga the DNA (?)

File: content/podcast/013-ryan.md [hunk 9]
```diff
@@ -0,0 +1,400 @@
...
... podcast episode a couple years ago now with Francois from interrupt who used to be at Pebble and we talked about this a lot of like —you talk about how people got into it where, especially on the lower end of when you're talking about like your PIC8s or whatever like, a lot of people were like double Es who were they were the one that was like software inclined, and they figured out how to get the PID Loop running on their PIC so they could have a better motor controller or something like that and they came up from that. Or you get the people from the opposite side who were doing backend services and they had done a little bit of electronic stuff before, so they got roped in to writing the microcontroller code. And you really do have these 2 very different worlds that are smashing together in the middle, where you get this really weird gray area of everything from like 32 bit microcontrollers up to very small microprocessors. There is no one style and there's no one right answer and particularly when everyone's cost optimizing or trying to get the most hard realtime out of everything that they're doing, there's no one size fit all approach; which is, whew, it's something for sure.
```
Comment (amanda-onevariable) [inline content/podcast/013-ryan.md]: I don't know exactly what interrupt refers to (if it should also be in the links) [here? ](https://interrupt.memfault.com/)
PIC8 
double Es

File: content/podcast/013-ryan.md [hunk 10]
```diff
@@ -0,0 +1,400 @@
...
...stems to be dumb, then you don't have to think about what is my acknowledgement window or you don't have to think about a response timeout because you just say, “Well, if I stop hearing messages for longer than 500 milliseconds the other device is faulted, or the line is broken, and we're done here,” because it should always send at every 100 millisecond or something like that. This sort of gets back to that PubSub versus request response where when you get people coming from a web background a lot of the times they're thinking in terms of like rest requests and things like that and you end up with this sort of like always RPC world where you send a request and you wait for a response versus when you get the more like double e brained people or people who have worked in control systems you go I don't want that because it's easier to just think of it in terms of a naive control system of there's inputs and outputs and we just are saying what we want and receiving back what they are. I was wondering, how much of the design of your system ends up being more like the web request response sort of world versus the control system “I want you to be this. I am this.” sort of back and forth?
```
Comment (amanda-onevariable) [inline content/podcast/013-ryan.md]: double e brained people ?

**Review by amanda-onevariable — COMMENTED** [review]
Here are my ?s inline

File: content/podcast/013-ryan.md (hunk 1 above)
Comment (jamesmunns) [reply content/podcast/013-ryan.md:157]: Noah Huesser: https://github.com/yatekii

File: content/podcast/013-ryan.md (hunk 2 above)
Comment (jamesmunns) [reply content/podcast/013-ryan.md:177]: Coaxial or "Coax" cable, like TV cable: https://en.wikipedia.org/wiki/Coaxial_cable

"Interlock": https://en.wikipedia.org/wiki/Interlock_(engineering)

File: content/podcast/013-ryan.md (hunk 3 above)
Comment (jamesmunns) [reply content/podcast/013-ryan.md]: This should be "MQTT" not "NQC". Also "8266" not "A266"

File: content/podcast/013-ryan.md (hunk 4 above)
Comment (jamesmunns) [reply content/podcast/013-ryan.md]: No worries, Robert is one of the other contributors to Stabilizer: https://github.com/jordens, probably talking about "miniconf": https://github.com/quartiq/miniconf

File: content/podcast/013-ryan.md (hunk 5 above)
Comment (jamesmunns) [reply content/podcast/013-ryan.md]: DB: "Database". "Influx": a specific kind of database: https://www.influxdata.com/

File: content/podcast/013-ryan.md (hunk 6 above)
Comment (jamesmunns) [reply content/podcast/013-ryan.md]: H7: STM32H7 - https://www.st.com/en/microcontrollers-microprocessors/stm32h7-series.html

Not sure who that Richard is.

ABC -> ADC "Analog to Digital Converter"

mini comp client -> "miniconf client"

File: content/podcast/013-ryan.md (hunk 7 above)
Comment (jamesmunns) [reply content/podcast/013-ryan.md:279]: Any of those work: https://en.wikipedia.org/wiki/I%C2%B2C

File: content/podcast/013-ryan.md (hunk 8 above)
Comment (jamesmunns) [reply content/podcast/013-ryan.md]: DSB -> DSP (Digital Signal Processing)

Scatter/Gather: https://en.wikipedia.org/wiki/Gather/scatter_(vector_addressing)

DMA: Direct Memory Access: https://en.wikipedia.org/wiki/Direct_memory_access

File: content/podcast/013-ryan.md (hunk 9 above)
Comment (jamesmunns) [reply content/podcast/013-ryan.md]: Yep, that interrupt.

PIC8 is a brand of microcontrollers: https://www.microchip.com/en-us/products/microcontrollers-and-microprocessors/8-bit-mcus

double-E or "EE": "Electrical Engineer(ing)"

File: content/podcast/013-ryan.md (hunk 10 above)
Comment (jamesmunns) [reply content/podcast/013-ryan.md]: "Electrical Engineering brained people" - people who think like electrical engineers (as opposed to people who write software)

**Pull request merged**